src/events/RelationsHelper.ts

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixClient, MatrixEvent, MatrixEventEvent, RelationType } from "matrix-js-sdk/src/matrix";
+import { Relations, RelationsEvent } from "matrix-js-sdk/src/models/relations";
+import { TypedEventEmitter } from "matrix-js-sdk/src/models/typed-event-emitter";
+
+import { IDestroyable } from "../utils/IDestroyable";
+
+export enum RelationsHelperEvent {
+    Add = "add",
+}
+
+interface EventMap {
+    [RelationsHelperEvent.Add]: (event: MatrixEvent) => void;
+}
+
+/**
+ * Helper class that manages a specific event type relation for an event.
+ * Just create an instance and listen for new events for that relation.
+ * Optionally receive the current events by calling emitCurrent().
+ * Clean up everything by calling destroy().
+ */
+export class RelationsHelper
+    extends TypedEventEmitter<RelationsHelperEvent, EventMap>
+    implements IDestroyable {
+    private relations: Relations;
```

Comment:
this should be optional since it may not be defined until the relations created event